admin/tabs/roles/RolesModel.ts

```diff
@@ -0,0 +1,280 @@
+import {RoleEditorModel} from '@xh/hoist/admin/tabs/roles/editor/RoleEditorModel';
+import {HoistRole} from '@xh/hoist/admin/tabs/roles/HoistRole';
+import {FilterChooserModel} from '@xh/hoist/cmp/filter';
+import {GridModel} from '@xh/hoist/cmp/grid';
+import * as Col from '@xh/hoist/cmp/grid/columns';
+import {div, li, ul} from '@xh/hoist/cmp/layout';
+import {HoistModel, LoadSpec, managed, persist, ReactionSpec, XH} from '@xh/hoist/core';
+import {RecordActionSpec} from '@xh/hoist/data';
+import {fmtDate} from '@xh/hoist/format';
+import {Icon} from '@xh/hoist/icon';
+import {bindable, makeObservable} from '@xh/hoist/mobx';
+import {pluralize} from '@xh/hoist/utils/js';
+import {compact, groupBy, mapValues} from 'lodash';
+import {action, observable} from 'mobx';
+import moment from 'moment/moment';
+
+export class RolesModel extends HoistModel {
+    static PERSIST_WITH = {localStorageKey: 'xhAdminRolesState'};
+
+    override persistWith = RolesModel.PERSIST_WITH;
+
+    readonly ACTIONS: RecordActionSpec[] = this.createRecordActions();
+
+    @managed readonly gridModel: GridModel = this.createGridModel();
+    @managed readonly filterChooserModel: FilterChooserModel = this.createFilterChooserModel();
+    @managed readonly roleEditorModel = new RoleEditorModel(this);
+
+    @observable.ref allRoles: HoistRole[] = [];
+
+    @bindable @persist groupByCategory = true;
+
+    get selectedRole(): HoistRole {
+        return this.gridModel.selectedRecord?.data as HoistRole;
+    }
+
+    constructor() {
+        super();
+        makeObservable(this);
+        this.addReaction(this.groupByCategoryReaction());
+    }
+
+    override async doLoadAsync(loadSpec: LoadSpec) {
+        try {
+            const {data} = await XH.fetchJson({loadSpec, url: 'roleAdmin/list'});
+            if (loadSpec.isStale) return;
+            this.setRoles(this.processRolesFromServer(data));
+        } catch (e) {
+            if (loadSpec.isStale) return;
+            XH.handleException(e);
+            this.clear();
+        }
+    }
+
+    selectRole(name: string) {
+        this.gridModel.selectAsync(name);
+    }
+
+    @action
+    setRoles(roles: HoistRole[]) {
+        this.allRoles = roles;
+        this.gridModel.loadData(roles);
+    }
+
+    @action
+    clear() {
+        this.allRoles = [];
+        this.gridModel.clear();
+    }
+
+    // -------------------------------
+    // Reactions
+    // -------------------------------
+
+    private groupByCategoryReaction(): ReactionSpec<boolean> {
+        const {gridModel} = this;
+        return {
+            track: () => this.groupByCategory,
+            run: groupByCategory => {
+                gridModel.setGroupBy(groupByCategory ? 'category' : null);
+                gridModel.setColumnVisible('category', !groupByCategory);
+                gridModel.autosizeAsync();
```

Comment:
Need to instruct gridModel to not persist grouping - otherwise restore defaults on grid gets you out of sync